python/src/ops.cpp

```diff
@@ -1367,9 +1371,9 @@ void init_ops(nb::module_& m) {
             dtype,
             s);
       },
-      "start"_a,
-      "stop"_a,
-      "step"_a = nb::none(),
+      "start"_a.noconvert(),
+      "stop"_a.noconvert(),
+      "step"_a.noconvert() = nb::none(),
```

Comment:
I added this because nanobind will otherwise implicitly convert `mx.arange(mx.array(1))` to work since scalar arrays are convertible to Python integers. It's not technically a bug.. but I think it's better to avoid implicit graph evaluations like that.